git-branchless-hook/src/lib.rs

```diff
@@ -42,6 +42,41 @@ pub use lib::core::rewrite::rewrite_hooks::{
     hook_skip_upstream_applied_commit,
 };
 
+/// A component that will find the hash for a given reference transaction OID.
+/// Reference transaction OIDs may be one of:
+///
+/// * An up-to-40-hex-digit hash string
+/// * A symbolic ref, in which case, it will start with "ref:"
+///
+/// See: https://github.com/git/git/commit/a8ae923f85da6434c3faf9c39719d6d5e5c77e65
+#[derive(Debug)]
+struct TransactionReferenceFinder<'a> {
+    repo: &'a Repo,
+}
+
+impl<'a> TransactionReferenceFinder<'a> {
+    fn new(repo: &'a Repo) -> Self {
+        Self { repo }
+    }
+
+    #[instrument]
+    /// Attempt to resolve the provided string as a transaction reference OID.
+    ///
+    /// If name starts with `ref:`, and the implementation has a reference lookup
+    /// mechanism, then the OID for the provided reference name will be
+    /// returned, or a failure, if no target was found.
+    ///
+    /// Otherwise, an attempt is made to parse `name` as an up-to-40-hex-digit hash,
```

Comment:
comment: I believe that this is imprecise, and the object ID will always be exactly 40 characters long (nothing about abbreviating it is mentioned in https://git-scm.com/docs/githooks#_reference_transaction), and, uh... the tests that use 6 characters instead of 40 characters were probably me being lazy because I knew it would work under the current implementation and I thought that it would be easier to read the tests that way 😬. But I can see now that it probably a bad idea insofar as one uses tests to determine the typical use of an interface.

You don't need to fix the tests, but, if we specify the exact format of the object ID in the comments/spec, it would be best to claim that we expect the object ID to be exactly 40 characters, or just don't specify that explicitly in the spec and maybe link to the `reference-transaction` hook documentation directly.